### PR TITLE
New cigs cartons

### PR DESCRIPTION
**Pull request by TorinoFermic**
## About The Pull Request
The PR adds one brand of cigs, cartons for all existing brands so they each hold 10 packets of cigs.

The new brand of cig contains adrenaline for now.

Up to discussion is the count of 6 cigs per parcket should be increased to 10 or 20 ?

WIP 

- Add these cartons and packets to vendors and loot tables for maintenance and deep maint.
- Adjust for the desc on the new brand line of cigs if needed.

## Why It's Good For The Game

Adds one brand of cigs containing a stim to grant stat bonuses.

## Pictures

image

## Changelog
:cl:
add: New brand of cigarettes with a stim effect inside them.
code: Cleaned the code to make possible to create new packets with various items preloaded instead being hardcoded inside populate_contents()
add: Updated vendors and guild for these cigcartons
add: added cartons to loot tables
/:cl:

:cl: SuicidalManiac
imageadd: Added cartons, one brand of packet.
/:cl:

### CODE REVIEW

**Comment by MadmanMartian** [comment]
Is cartoon some meme on carton?

**Comment by TorinoFermic** [comment]
> 
> 
> Is cartoon some meme on carton?

Oh man, I misspell them. I am fixing them now

**Comment by mattroks101** [comment]
That sprite looks super goofy.

**Comment by MadmanMartian** [comment]
something, something, wide packet walk

File: code/game/objects/items/weapons/storage/fancy.dm [hunk 1]
```diff
@@ -197,6 +198,62 @@
 	. = ..()
 	fill_cigarre_package(src, list("fuel" = 15))
 
+/obj/item/weapon/storage/fancy/cigarettes/homeless
+	name = "\improper Nomads packet"
+	desc = "A packet of six Nomads cigarettes. Nomads's Extra strong for when your life is more extra hard"
```
Comment (ACCount12) [inline code/game/objects/items/weapons/storage/fancy.dm:203]: Reenglish?

File: code/game/objects/items/weapons/storage/fancy.dm [hunk 2]
```diff
@@ -197,6 +198,62 @@
...
+		icon_state = "[initial(icon_state)]"
+
+/obj/item/weapon/storage/fancy/cigcartons/populate_contents()
+	for(var/i in 1 to storage_slots)
+		new item_obj(src)
+	update_icon()
+
+/obj/item/weapon/storage/fancy/cigcartons/dromedaryco
+	name = "carton of dromedaryco cigarettes"
+	desc = "A box containing 10 packets of dromedarycos cigarettes."
```
Comment (ACCount12) [inline code/game/objects/items/weapons/storage/fancy.dm:238]: The other company names are capitalized. Why this one isn't?

Comment (TorinoFermic) [reply code/game/objects/items/weapons/storage/fancy.dm:238]: Copy paste straight out of path name

File: code/game/objects/items/weapons/storage/fancy.dm (hunk 1 above)
Comment (TorinoFermic) [reply code/game/objects/items/weapons/storage/fancy.dm:203]: WIP until someone brings up a good desc, btw it was from SuicidalSquirrel.

File: code/game/objects/items/weapons/storage/fancy.dm (hunk 2 above)
Comment (mattroks101) [reply code/game/objects/items/weapons/storage/fancy.dm:238]: Should still be capitalized. This is the name of a company isn't it?

**Comment by TorinoFermic** [comment]
Removing the discussion needed tag since I haven't heard anything new about the adrenaline I added in the new brand of cigs.

File: code/game/machinery/vending.dm [hunk 3]
```diff
@@ -1082,11 +1082,24 @@
...
-	contraband = list(/obj/item/clothing/mask/smokable/cigarette/cigar = 4, /obj/item/weapon/flame/lighter/zippo = 4,)
-	premium = list(/obj/item/weapon/storage/fancy/cigar = 5,/obj/item/weapon/storage/fancy/cigarettes/killthroat = 5 )
-	prices = list(/obj/item/clothing/mask/smokable/cigarette/cigar = 200, /obj/item/weapon/storage/fancy/cigarettes = 100 ,/obj/item/weapon/storage/box/matches = 10,/obj/item/weapon/flame/lighter/random = 5,
-				/obj/item/weapon/flame/lighter/zippo = 250)
+	products = list(/obj/item/weapon/storage/fancy/cigarettes = 10,
+					/obj/item/weapon/storage/fancy/cigcartons = 5,
+					/obj/item/weapon/storage/box/matches = 10,
+					/obj/item/weapon/flame/lighter/random = 4)
+
+	contraband = list(/obj/item/clothing/mask/smokable/cigarette/cigar = 4,
```
Comment (SpadesSlick) [inline code/game/machinery/vending.dm:1090]: Both "contraband" and "premium" selections are something we should move away from. We don't need coin functionality or hacking vendors if we want some sort of economy. 

Just adjust the price for new cigarettes.

File: code/modules/cargo/packs.dm [hunk 4]
```diff
@@ -1475,23 +1496,14 @@ var/list/all_supply_groups = list("Operations","Security","Hospitality","Enginee
...
+	num_contained = 2
+	contains = list(/obj/item/weapon/storage/fancy/cigcartons,
+				/obj/item/weapon/storage/fancy/cigcartons/dromedaryco,
+				/obj/item/weapon/storage/fancy/cigcartons/killthroat,
+				/obj/item/weapon/storage/fancy/cigcartons/homeless)
+	name = "Cigarettes Cartons"
+	cost = 1200
+	crate_name = "Cigarettes Cartons Crate"
+	containertype = /obj/structure/closet/crate
+	group = "Supply"
```
Comment (SpadesSlick) [inline code/modules/cargo/packs.dm:1509]: Endline.

File: code/game/objects/items/weapons/storage/fancy.dm [hunk 5]
```diff
@@ -197,6 +198,62 @@
 	. = ..()
 	fill_cigarre_package(src, list("fuel" = 15))
 
+/obj/item/weapon/storage/fancy/cigarettes/homeless
+	name = "\improper Nomads packet"
+	desc = "A packet of six Nomads cigarettes. Nomads's Extra strong for when your life is more extra hard"
+	icon_state = "Cpacket"
+	item_state = "Cpacket"
+
+/obj/item/weapon/storage/fancy/cigarettes/homeless/Initialize()
```
Comment (SpadesSlick) [inline code/game/objects/items/weapons/storage/fancy.dm:207]: So I see why this is done this way, but we should do it properly with cig subtypes with different reagents inside them.

**Comment by drexample** [comment]
WIDE

**Comment by TorinoFermic** [comment]
Working to finish today on 5 hour energy drink.

**Comment by Kurgis** [comment]
> 
> 
> Working to finish today on 5 hour energy drink.

Don't do that.

**Pull request merged**